### PR TITLE
Improve filter button

### PR DESCRIPTION
**Pull request by msecret**
This moves the filter button to the right side of the filter bar. This
was done so the filter button UI was attached to the actual filter
UI, so the button affords closing and opening the side bar.

This iteration doesn't include any animation. Animation would require
some extra work.

Fixes https://github.com/18F/openFEC-web-app/issues/293

### CODE REVIEW

**Comment by msecret** [comment]
This is how it's looking at various breakpoints. @jenniferthibault @noahmanger, the position of the filter buttons breaks at the smallest breakpoint when its minified. It's running into the header below it.

screen shot 2015-07-01 at 4 53 03 pm
screen shot 2015-07-01 at 4 53 20 pm
screen shot 2015-07-01 at 4 53 37 pm
screen shot 2015-07-01 at 4 53 52 pm
screen shot 2015-07-01 at 4 54 01 pm
screen shot 2015-07-01 at 4 54 11 pm
screen shot 2015-07-01 at 4 54 16 pm

**Comment by jenniferthibault** [comment]
@msecret I'm working through some other tweaks to different elements on other screen sizes you've shown, but: 
A) Is this set of changes to the mobile view possible?; 
B) Is this kind of markup helpful?

artboard 20

**Comment by noahmanger** [comment]
This looks way better. A couple other ideas:

1. Would love for this to default to closed on mobile so you see the data first.
2. If that's the case, we may need an indicator that filters are applied.

**Comment by jenniferthibault** [comment]
Next up: Show filters on mobile screens. Same philosophy for the changes you see here as in "hide filter" above.
artboard 20

**Comment by jenniferthibault** [comment]
Medium and XL screens:

artboard 23

artboard 22


Glad to talk through anything that throws a monkey wrench into something that I haven't noticed!

**Comment by jenniferthibault** [comment]
I was confusing myself with all the descriptions of how the menu worked. I hope this helps to clarify!
artboard 24

**Comment by noahmanger** [comment]
On small screens I'm worried that the icon switching sides with the text might be kind of a weird jump. Is there a reason you had it like that?

**Comment by jenniferthibault** [comment]
THANK YOU @noahmanger ! It made sense in my head when I was making it, but I didn't pause to review it when finished. For small screens, the icon is now on the left so that people read it (and hopefully get the signal) before they have to read the text. For large screens, since the text is moving left-right, the icon is on the right so that the words disappear & reveal in-sync with the animation. Does that make sense, or should they all be on the same side?  cc @msecret — sorry for the confusion here.

artboard 24 copy

**Comment by noahmanger** [comment]
Nice. Yeah I think it makes sense to be on the right on large screens. I like it!

**Comment by msecret** [comment]
@noahmanger @jenniferthibault , I just updated this, here are some new screens.

screen shot 2015-07-07 at 3 27 11 pm
screen shot 2015-07-07 at 3 27 17 pm
screen shot 2015-07-07 at 3 27 29 pm
screen shot 2015-07-07 at 3 27 42 pm

**Comment by msecret** [comment]
Still left to do:
- Mobile animation for filter slide up. I'm having trouble with this, not 100% we should do it right now.
- Fix staggering fields problem on tablet. I think I might want to put this in a different PR has it's pretty separate from this issue.

File: static/styles/_components/_side-panel.scss [hunk 1]
```diff
@@ -23,24 +23,144 @@
 }
 
 .side-panel {
```
Comment (noahmanger) [inline static/styles/_components/_side-panel.scss]: This is my fault for the way this was originally implemented, but the glossary also has the `.side-panel` class, so all of these style changes break it. We could either move them to `.side-panel--left` or remove the `.side-panel` class from the glossary. (They started out as similarly functioning things, so made sense for them to share some styles, but they're no pretty different so maybe it makes more sense to break them out).

File: static/styles/_components/_side-panel.scss [hunk 2]
```diff
@@ -23,24 +23,144 @@
...
-  padding: 2rem;
-  top: 0;
+  background: $lightest-gray;
+  border-color: $light-gray;
+  border-style: solid;
+  border-bottom-width: 1px;
+  border-top-width: 0;
   bottom: 0;
+  //padding: 1rem;
```
Comment (noahmanger) [inline static/styles/_components/_side-panel.scss]: Does this need to stay in here?

File: static/styles/_components/_side-panel.scss [hunk 3]
```diff
@@ -23,24 +23,144 @@
...
+  position: absolute;
+  top: 0;
+  width: 100%;
+
+  h4 {
+    margin-top: 1em;
+  }
+
+  &.side-panel--open {
```
Comment (noahmanger) [inline static/styles/_components/_side-panel.scss]: Does this need to be nested or can it be moved up a level?

File: static/styles/_components/_side-panel.scss [hunk 4]
```diff
@@ -23,24 +23,144 @@
...
+    float: left;
+    margin-left: -17%;
+    position: static;
+    width: 17%;
+
+    .side-panel__box {
+      position: relative;
+    }
+
+    .side-panel__toggle.side-panel__toggle {
```
Comment (noahmanger) [inline static/styles/_components/_side-panel.scss]: Is this a typo?

File: templates/candidates.html [hunk 5]
```diff
@@ -11,27 +11,15 @@
...
-        {% include 'partials/filters/parties.html' %}
-        {% include 'partials/filters/states.html' %}
-        {% include 'partials/filters/districts.html' %}
-
-      <div class="field">
-        <button type="submit" class="primary">Apply filters</button>
-      </div>
-    </form>
-  </div>
+  {% set filters = [
```
Comment (noahmanger) [inline templates/candidates.html]: Yessss. Love this.

File: static/styles/_components/_side-panel.scss (hunk 4 above)
Comment (msecret) [inline static/styles/_components/_side-panel.scss]: Unfortunately, no. It's a big of a specificity hack to get around having to use `!important`.

Comment (noahmanger) [inline static/styles/_components/_side-panel.scss]: Oh cool. I had no idea.

File: static/js/modules/tables.js [hunk 6]
```diff
@@ -165,6 +165,7 @@ function initTable($table, $form, baseUrl, columns) {
   var $paging = $(api.table().container()).find('.results-info--top');
   $paging.prepend($('#filter-toggle'));
```
Comment (noahmanger) [inline static/js/modules/tables.js]: This line needs to be removed. It's moving the button.

**Pull request merged**